sns/client.bal

```
@@ -0,0 +1,501 @@
+    final http:Client amazonSNSClient;
+
+    # Initializes the connector.
+    #
+    # + configuration - Configuration for the connector
+    # + httpClientConfig - HTTP Configuration
+    # + return - `http:Error` in case of failure to initialize or `null` if successfully initialized
+    public isolated function init(ConnectionConfig configuration, http:ClientConfiguration httpClientConfig = {}) returns error? {
+        self.accessKeyId = configuration.credentials.accessKeyId;
+        self.secretAccessKey = configuration.credentials.secretAccessKey;
+        self.securityToken = (configuration?.credentials?.securityToken is string) ? <string>(configuration?.credentials?.securityToken) : ();
+        self.region = configuration.region;
+        self.amazonHost = "sns." + self.region + ".amazonaws.com";
+        string baseURL = "https://" + self.amazonHost;
+        check checkCredentials(self.accessKeyId, self.secretAccessKey);
+        self.amazonSNSClient = check new (baseURL, httpClientConfig);
+    }
+
+    # Create a topic.
+    #
+    # + name - Name of topic
+    # + attributes - Topic attributes
+    # + tags - Tags
+    # + return - Created topic ARN on success else an `error`
+    @display {label: "Create Topic"}
+    isolated remote function createTopic(@display {label: "Topic Name"} string name, 
+                                         @display {label: "Topic Attributes"} TopicAttributes? attributes = (), 
+                                         @display {label: "Tags"} map<string>? tags = ()) 
+                                         returns @tainted @display {label: "Created Topic ARN"} CreateTopicResponse|error {
+        map<string> parameters = {};
+        parameters = createQueryString("CreateTopic", parameters);
+        parameters["Name"] = name;
+        parameters = check addTopicOptionalParameters(parameters, attributes, tags);
+        http:Request|error request = self.generateRequest(self.createPayload(parameters));
```

```suggestion
        http:Request request = check self.generateRequest(self.createPayload(parameters));
```
Error is not handled. Please check other functions.